File: Listview/app/src/main/java/com/example/listview/MainActivity.java
package com.example.listview;

import android.os.Bundle;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ListView listV;
    List<MessageModel> listmsg;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        define();
        fillList();

    }

    public void define(){
        listV = findViewById(R.id.listview);
    }

public void fillList(){
    listmsg = new ArrayList<>();
    MessageModel m1 = new MessageModel("Hello I am Jane", "Jane", R.drawable.ic_launcher_background);
    MessageModel m2 = new MessageModel("Hello I am Bear", "Bear", R.drawable.ic_launcher_foreground);
    listmsg.add(m1); listmsg.add(m2);



}
}

File: activityLifecycle/app/src/main/java/com/busrayasar/activitylifecycle/MainActivity.kt
package com.busrayasar.activitylifecycle

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        println("onCreate is called. ")
    }

    override fun onStart() {
        super.onStart()
        println("onStart is called. ")
    }

    override fun onResume() {
        super.onResume()
        println("onResume is called. ")
    }

    override fun onPause() {
        super.onPause()
        println("onPause is called. ")
    }

    override fun onStop() {
        super.onStop()
        println("onStop is called. ")
    }

    override fun onDestroy() {
        super.onDestroy()
        println("onDestroy is called. ")
    }

    //aktiviteler arası veri yollayalım
    fun changePage(view: View){
        //kullanıcı verisi alalım
        //*******************************************************************************
        val editText = findViewById<EditText>(R.id.editText) // editTexti tanımadığı için bunu ekledim
        //****************************************************************************************
        val userInput = editText.text.toString()
        val intent = Intent(applicationContext, MainActivity2::class.java)
        intent.putExtra("Set value:", userInput)
        startActivity(intent) // bununla sayfaya geçiş çalışır
        finish() //bu fonksiyonla aktivite kapatılır (destroy)
        //SecondActivity::class.java -> ikinci aktiviteye git
    }


}